bite-size-notes/week-7-a-kalman-filter.md

```diff
@@ -0,0 +1,129 @@
+
+$$\mu_k^{pred}=A\mu_{k-1}+Bu_{k-1}$$
+
+Furthermore, the current covariance is by definition
+
+$$\Sigma_k^{pred}=\mathbb{E}[(x_k-\mu_k^{pred})(x_k-\mu_k^{pred})^T]$$
+
+Where $x_k$ is the true state at the current time step. This is unknown but what is important here is that the above properties can be calculated as functions of the previous values and the dynamics.
+
+$$x_{k}=Ax_{k-1}+Bu_{k-1}$$
+
+Plugging the equations for $x_k$ and $\mu_k$ into the equation for $\Sigma_k^{pred}$ and simplifying yields a recursive equation for the new covariance prediction in terms of the previous covariance.
+
+$$\Sigma_k^{pred}=A\Sigma_{k-1}a^T+Q$$
+
+Notice that these recursive equations are calculating **predictions** of the current estimated state Gaussian. In order to correct these predictions using observations some update equation must account for any error between the observation and the predicted output. For $\mu_k$ this is relatively straighforward. Similar to a controls problem, the error multiplied by some gain  can be used to drive the predicted mean towards the true mean.
+
+$$\mu_k=\mu_k^{pred}+K_k(y_k-C\mu_k^{pred})$$
+
+Rearranging the above equation yields the update equation previously found in the implementation section. 
+
+To find the update equation for the covariance as well as an update equation for $K_k$, consider the definition of the true covariance.
+
+$$\Sigma_k=\mathbb{E}[(x_k-\mu_k)(x_k-\mu_k)^T]$$
+
+$\mu_k$ can be defined as above, but $y_k$ can be instead written in terms of $x_k$ using $y_k=C(x_k+v_k)$, the definition of the output from the linear dynamics.
+
+$$m_k=(I-K_kC)\mu_k^{pred}+K_kC(x_k+v_k)$$
+
+Plugging this into the previous definition of the true covariance and rearranging yields the following equation for the true covariance.
+
+$$\Sigma_k=(I-K_kC)\mathbb{E}[(x_k-\mu_k^{pred})(x_k-\mu_k^{pred})^T] (I-K_kC)^T+K_k \mathbb{E}[v_kv_k^T] K_k^T$$
+
+However, notice that the expected value here is just $\Sigma_k^{pred}$, or the predicted covariance at the current time step, and that the expected value of the squared $v_k$ is just the covariance $R$ of $v$. Plugging this into the above equation yields the equation that corrects the predicted covariance.
+
+$$\Sigma_k=(I-K_kC)\Sigma_k^{pred}(I-K_kC)^T+K_kRK_k^T$$
+
+This equation determines the covariance at the current time step. In order to have a more accurate system, $K_k$ should be choosen such that the covariance, or amount of variation between variables, is minimized. Rather than directly minimizing the matrix, the trace of the above equation is instead minimized to find $K_k$. This yields the optimal gain matrix $K_k^*$.
+
+$$K_k^*=\underset{K_k}{argmin}Tr(\Sigma_k)$$
+
+Solving the above minimization problem provides the following equation for $K_k$
+
+$$K_k^*=\Sigma_k^{pred}C^T(C\Sigma_k^{pred}C^T+R)^{-1}$$
+
+This equation produces the optimal gain matrix for each time step using the predicted covariance.
+
+## Conclusion
```

Comment:
You may put the last sentence in the preceding body paragraph and end it by summarizing the previously introduced